### DISCUSSION TITLE
Isolating grains with high angle boundaries

### DISCUSSION THREAD

**Discussion by tscuseria** [mtex-toolbox/mtex#1824] [Ask Anything]
I am looking to acquire grain size of grains with high angle grain boundaries.
```matlab
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',[2*degree]);
grains = grains(grains.grainSize > 5);
ebsd=ebsd(grains);
[grains,ebsd.grainId] = calcGrains(ebsd,'angle',[2*degree]);
grains = smooth(grains,5);

gB=grains.boundary('Austenite','Austenite')
mAngle = gB.misorientation.angle./ degree;
[~,id] = histc(mAngle,[2 15 120]);

plot(ebsd('Austenite'),ebsd('Austenite').orientations,'FaceAlpha',0.75,'figSize','large')
hold on
plot(gB(id==1),'linecolor','k','linewidth',2,'DisplayName','2-15^\circ')
hold on
plot(gB(id==2),'linecolor','r','linewidth',2,'DisplayName','>15^\circ')
```
![IPF_GB+SubGB_2deg](https://github.com/mtex-toolbox/mtex/assets/123943246/4ec63bff-e15a-466d-b58c-97de30163f86)

I'm not sure how to isolate these grains once I get to this step.

I tried this:
```
gB=grains.boundary('Austenite','Austenite')
newgrains= grains(unique(gB(gB.misorientation.angle>15*degree).grainId(:)))
plot(newgrains)
```
![image](https://github.com/mtex-toolbox/mtex/assets/123943246/fd69c957-b987-4f9a-9c98-aa0b7c0e93c7)

It obviously misses a lot of the red boundaries plotted above?

If I threshold in the original step for 15 degrees instead of 2 degrees, MTEX misses a lot of the original grain boundaries from the first plot, even though they were above 15 degrees?:

![IPF_GB_15deg](https://github.com/mtex-toolbox/mtex/assets/123943246/e9ed047b-6a8e-4f17-83ad-c462363c0c22)

Threshold for 15 deg:
![IPF_GB_15deg_Inset](https://github.com/mtex-toolbox/mtex/assets/123943246/796b22ff-b591-4387-9346-a40c473e1f10)

Threshold for 2 deg:
![IPF_GB+SubGB_2deg_Inset](https://github.com/mtex-toolbox/mtex/assets/123943246/8e2ef768-6fd3-4fbd-9e1d-8bc95452f558)

**Comment by ralfHielscher** [comment]
Hi,
I did not really understood what you want to accomplish. You tried two variants:

1. Grains are surrounded by grain boundary with angle > 2*degree. You selected only those grains that have at least one segment with angle >= 15 degree.
In short: all segments have misorientation angle > 2 degrees at least one has misorientation angle > 15 degrees
2. All segments have misorientation angle >= 15 degrees

What else are you looking for?
Ralf.

**Reply by tscuseria** [reply]
Hi Ralf. After some further thinking I think all I need is a refinement of this step: 

gB=grains.boundary('Austenite','Austenite')
newgrains= grains(unique(gB(gB.misorientation.angle>15*degree).grainId(:)))
plot(newgrains)

Where in this case, I don't want grains with **at least one** segment with >=15 deg, but grains that **only** contain segments >=15 deg.

**Reply by ralfHielscher** [reply]
But isn't that what option (2) gives you?
